src/Response/RedirectResponse.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Drupal\commerce\Response;
+
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Routing\TrustedRedirectResponse;
+
+/**
+ * Provides an exception that represents the need for an HTTP redirect.
+ */
+class RedirectResponse extends \LogicException {
```

Comment:
I wonder if we can name this class better. Drupal suffixes all Exceptions with Exception. And even our own subscriber is clarifying that the redirect response is actually an exception: "Check for RedirectResponse which extends Exception.".

Maybe we can call it NeedsRedirectException? Or just RedirectException?